### ISSUE TITLE
Problem with the use of `multiprocessing` in `NESTTOOLS.detect_nest_version()`

### ISSUE THREAD

**Issue by WillemWybo** [nest/nestml#814]
There is an issue with the use of `multiprocessing` in `NESTTOOLS.detect_nest_version()`. In tests, this function needs to be called to set a global variable `nest_version` that determines if the test is to be run, but calling this function without the `if __name__=="__main__` guard results in an endless instantiation of new processes on some systems, as per the error below:

```
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/Users/wybo/mambaforge3/envs/nest_devel/lib/python3.9/multiprocessing/spawn.py", line 116, in spawn_main
    exitcode = _main(fd, parent_sentinel)
  File "/Users/wybo/mambaforge3/envs/nest_devel/lib/python3.9/multiprocessing/spawn.py", line 125, in _main
    prepare(preparation_data)
  File "/Users/wybo/mambaforge3/envs/nest_devel/lib/python3.9/multiprocessing/spawn.py", line 236, in prepare
    _fixup_main_from_path(data['init_main_from_path'])
  File "/Users/wybo/mambaforge3/envs/nest_devel/lib/python3.9/multiprocessing/spawn.py", line 287, in _fixup_main_from_path
    main_content = runpy.run_path(main_path,
  File "/Users/wybo/mambaforge3/envs/nest_devel/lib/python3.9/runpy.py", line 288, in run_path
    return _run_module_code(code, init_globals, run_name,
  File "/Users/wybo/mambaforge3/envs/nest_devel/lib/python3.9/runpy.py", line 97, in _run_module_code
    _run_code(code, mod_globals, init_globals,
  File "/Users/wybo/mambaforge3/envs/nest_devel/lib/python3.9/runpy.py", line 87, in _run_code
    exec(code, run_globals)
  File "/Users/wybo/Code/NEST/nestml/tests/nest_tests/compartmental_model_test.py", line 75, in <module>
    class CMTest(unittest.TestCase):
  File "/Users/wybo/Code/NEST/nestml/tests/nest_tests/compartmental_model_test.py", line 238, in CMTest
    @pytest.mark.skipif(get_nest_version().startswith("v2"),
  File "/Users/wybo/Code/NEST/nestml/tests/nest_tests/compartmental_model_test.py", line 72, in get_nest_version
    nest_version = NESTTools.detect_nest_version()
  File "/Users/wybo/Code/NEST/nestml/pynestml/codegeneration/nest_tools.py", line 72, in detect_nest_version
    p = mp.Pool(processes=1)
  File "/Users/wybo/mambaforge3/envs/nest_devel/lib/python3.9/multiprocessing/context.py", line 119, in Pool
    return Pool(processes, initializer, initargs, maxtasksperchild,
  File "/Users/wybo/mambaforge3/envs/nest_devel/lib/python3.9/multiprocessing/pool.py", line 212, in __init__
    self._repopulate_pool()
  File "/Users/wybo/mambaforge3/envs/nest_devel/lib/python3.9/multiprocessing/pool.py", line 303, in _repopulate_pool
    return self._repopulate_pool_static(self._ctx, self.Process,
  File "/Users/wybo/mambaforge3/envs/nest_devel/lib/python3.9/multiprocessing/pool.py", line 326, in _repopulate_pool_static
    w.start()
  File "/Users/wybo/mambaforge3/envs/nest_devel/lib/python3.9/multiprocessing/process.py", line 121, in start
    self._popen = self._Popen(self)
  File "/Users/wybo/mambaforge3/envs/nest_devel/lib/python3.9/multiprocessing/context.py", line 284, in _Popen
    return Popen(process_obj)
  File "/Users/wybo/mambaforge3/envs/nest_devel/lib/python3.9/multiprocessing/popen_spawn_posix.py", line 32, in __init__
    super().__init__(process_obj)
  File "/Users/wybo/mambaforge3/envs/nest_devel/lib/python3.9/multiprocessing/popen_fork.py", line 19, in __init__
    self._launch(process_obj)
  File "/Users/wybo/mambaforge3/envs/nest_devel/lib/python3.9/multiprocessing/popen_spawn_posix.py", line 42, in _launch
    prep_data = spawn.get_preparation_data(process_obj._name)
  File "/Users/wybo/mambaforge3/envs/nest_devel/lib/python3.9/multiprocessing/spawn.py", line 154, in get_preparation_data
    _check_not_importing_main()
  File "/Users/wybo/mambaforge3/envs/nest_devel/lib/python3.9/multiprocessing/spawn.py", line 134, in _check_not_importing_main
    raise RuntimeError('''
RuntimeError: 
        An attempt has been made to start a new process before the
        current process has finished its bootstrapping phase.

        This probably means that you are not using fork to start your
        child processes and you have forgotten to use the proper idiom
        in the main module:

            if __name__ == '__main__':
                freeze_support()
                ...

        The "freeze_support()" line can be omitted if the program
        is not going to be frozen to produce an executable.
```

**Comment by pnbabu** [comment]
The way a new process is created in macOS is different from that of Unix-based systems. The default start method for a new process supported by the `multiprocessing` library from Python 3.8 in macOS is `spawn` while it is `fork` in Unix. The different start methods and what they mean can be found [here](https://docs.python.org/3.10/library/multiprocessing.html#contexts-and-start-methods). 

The multiprocessing code must be guarded by `if __name__=="__main__"` for the `spawn` start method to safely import the main module as suggested in [this section](https://docs.python.org/3/library/multiprocessing.html#the-spawn-and-forkserver-start-methods) of the docs. However, in order to circumvent this error in the tests, PR #821 moves the `NESTTools.detect_nest_version()` out of the global scope.

**Comment by clinssen** [comment]
This should have been fixed in #842. Please feel free to re-open if there is still an issue. Cheers!
